x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -322,3 +323,19 @@ lookupCommand
 inlinestatsCommand
     : DEV_INLINESTATS stats=aggFields (BY grouping=fields)?
     ;
+
+joinCommand
+    : type=(DEV_JOIN_LOOKUP | DEV_JOIN_LEFT | DEV_JOIN_RIGHT)? DEV_JOIN joinTarget joinCondition
+    ;
+
+joinTarget
+    : index=identifier (AS alias=identifier)?
+    ;
+
+joinCondition
+    : ON joinPredicate (COMMA joinPredicate)*
```

Comment:
ON a is the same as USING a.
Moving forward USING should be converted in an expression (after the field resolution happens since we don't have qualifiers and thus the name is not enough).